File: Source/Obsequy.Web/App/Forms/providerRegistrationForm.js


angular.module('providerRegistration-form', [])
	.factory('ProviderRegistrationForm', ['ProviderMember', function (ProviderMember) {
		var ProviderRegistrationForm = function (properties) {

 			angular.extend(this, {

 				member: new ProviderMember(),
 				password: undefined,
 				confirmPassword: undefined,
 				hasAcceptedEULA: undefined,

 				update: function (properties) {
 					properties = properties || {};

 					// update properties
 					this.member.update(properties.member);
 					this.password = properties.password;
 					this.confirmPassword = properties.confirmPassword;
 					this.hasAcceptedEULA = properties.hasAcceptedEULA;
 				},

 				toJSON: function () {
 					return {
 						member: this.member.toJSON(),
 						password: this.password,
 						confirmPassword: this.confirmPassword,
 						hasAcceptedEULA: this.hasAcceptedEULA
 					};
 				}
 			});

			this.update(properties);
		};

		return ProviderRegistrationForm;
	}]);

File: Source/Obsequy.Model/Form/ProviderRegistrationForm.cs
using FluentValidation.Results;
using Obsequy.Utility;

namespace Obsequy.Model
{
	public class ProviderRegistrationForm
	{
		#region Properties

		public ProviderMember Member { get; set; }
		public string Password { get; set; }
		public string ConfirmPassword { get; set; }
		public bool? HasAcceptedEULA { get; set; }

		#endregion

		public ProviderRegistrationForm()
		{
			this.Member = new ProviderMember();
		}

		public ValidationResult Validate(AccountSession accountSession, ValidationMode validationMode)
		{
			return ((new ProviderRegistrationFormValidator(accountSession, validationMode)).Validate(this) as ValidationResult);
		}
	}
}
